File: im-backend/modules/local_life/service/SemanticSearchService.java
package com.im.backend.modules.local_life.service;

import com.im.backend.modules.local_life.dto.*;

import java.util.List;

/**
 * 语义搜索服务接口
 * 处理自然语言POI搜索请求
 * 
 * @author IM Development Team
 * @since 2026-03-28
 */
public interface SemanticSearchService {

    /**
     * 执行自然语言搜索
     *
     * @param request 搜索请求
     * @param userId 用户ID
     * @return 搜索结果列表
     */
    List<SemanticSearchResultDTO> search(NaturalLanguageSearchRequestDTO request, Long userId);

    /**
     * 执行语义搜索（基于向量的相似度搜索）
     *
     * @param query 查询文本
     * @param semanticVector 语义向量
     * @param limit 返回数量
     * @return 搜索结果列表
     */
    List<SemanticSearchResultDTO> semanticSearch(String query, String semanticVector, int limit);

    /**
     * 获取搜索建议
     *
     * @param query 部分输入
     * @param latitude 纬度
     * @param longitude 经度
     * @return 建议列表
     */
    List<String> getSearchSuggestions(String query, Double latitude, Double longitude);

    /**
     * 获取热门搜索
     *
     * @param latitude 纬度
     * @param longitude 经度
     * @param limit 数量限制
     * @return 热门搜索词列表
     */
    List<String> getHotSearches(Double latitude, Double longitude, int limit);

    /**
     * 获取搜索历史
     *
     * @param userId 用户ID
     * @param limit 数量限制
     * @return 搜索历史列表
     */
    List<String> getSearchHistory(Long userId, int limit);

    /**
     * 清除搜索历史
     *
     * @param userId 用户ID
     */
    void clearSearchHistory(Long userId);

    /**
     * 基于会话上下文进行搜索
     *
     * @param sessionId 会话ID
     * @param query 当前查询
     * @return 搜索结果
     */
    List<SemanticSearchResultDTO> searchWithContext(String sessionId, String query);

    /**
     * 获取零结果时的推荐
     *
     * @param query 原查询
     * @return 推荐结果
     */
    List<SemanticSearchResultDTO> getZeroResultRecommendations(String query);
}

File: im-mobile/lib/modules/local_life/services/semantic_search_service.dart
import 'dart:convert';
import '../../../core/services/http_client.dart';
import '../models/semantic_search_result.dart';
import '../providers/semantic_search_provider.dart';

/// 语义搜索服务
/// 处理自然语言POI搜索相关的API调用
/// 
/// Author: IM Development Team
/// Since: 2026-03-28
class SemanticSearchService {
  static final SemanticSearchService _instance = SemanticSearchService._internal();
  factory SemanticSearchService() => _instance;
  SemanticSearchService._internal();

  final HttpClient _httpClient = HttpClient();

  /// 执行语义搜索
  Future<List<SemanticSearchResult>> search({
    required String query,
    double? latitude,
    double? longitude,
    int radius = 3000,
    String? sessionId,
    bool isVoiceInput = false,
  }) async {
    final response = await _httpClient.post(
      '/api/v1/semantic-search/search',
      data: {
        'query': query,
        'latitude': latitude,
        'longitude': longitude,
        'radius': radius,
        'sessionId': sessionId,
        'isVoiceInput': isVoiceInput,
      },
    );

    if (response.statusCode == 200) {
      final data = response.data['data'] as List;
      return data.map((json) => SemanticSearchResult.fromJson(json)).toList();
    }

    throw Exception('搜索失败: ${response.statusMessage}');
  }

  /// 识别搜索意图
  Future<SearchIntent> recognizeIntent(String query, {String? sessionId}) async {
    final response = await _httpClient.post(
      '/api/v1/semantic-search/intent',
      queryParameters: {
        'query': query,
        'sessionId': sessionId,
      },
    );

    if (response.statusCode == 200) {
      return SearchIntent.fromJson(response.data['data']);
    }

    throw Exception('意图识别失败');
  }

  /// 获取搜索建议
  Future<List<String>> getSuggestions({
    required String query,
    double? latitude,
    double? longitude,
  }) async {
    final response = await _httpClient.get(
      '/api/v1/semantic-search/suggestions',
      queryParameters: {
        'query': query,
        'latitude': latitude,
        'longitude': longitude,
      },
    );

    if (response.statusCode == 200) {
      final data = response.data['data'] as List;
      return data.cast<String>();
    }

    return [];
  }

  /// 获取热门搜索
  Future<List<String>> getHotSearches({
    double? latitude,
    double? longitude,
    int limit = 10,
  }) async {
    final response = await _httpClient.get(
      '/api/v1/semantic-search/hot-searches',
      queryParameters: {
        'latitude': latitude,
        'longitude': longitude,
        'limit': limit,
      },
    );

    if (response.statusCode == 200) {
      final data = response.data['data'] as List;
      return data.cast<String>();
    }

    return [];
  }

  /// 获取搜索历史
  Future<List<String>> getSearchHistory({int limit = 20}) async {
    final response = await _httpClient.get(
      '/api/v1/semantic-search/history',
      queryParameters: {'limit': limit},
    );

    if (response.statusCode == 200) {
      final data = response.data['data'] as List;
      return data.cast<String>();
    }

    return [];
  }

  /// 清除搜索历史
  Future<void> clearSearchHistory() async {
    await _httpClient.delete('/api/v1/semantic-search/history');
  }

  /// 基于上下文的搜索
  Future<List<SemanticSearchResult>> searchWithContext({
    required String sessionId,
    required String query,
  }) async {
    final response = await _httpClient.post(
      '/api/v1/semantic-search/context-search',
      queryParameters: {
        'sessionId': sessionId,
        'query': query,
      },
    );

    if (response.statusCode == 200) {
      final data = response.data['data'] as List;
      return data.map((json) => SemanticSearchResult.fromJson(json)).toList();
    }

    return [];
  }

  /// 获取澄清问题
  Future<List<String>> getClarificationQuestions(String query) async {
    final response = await _httpClient.post(
      '/api/v1/semantic-search/clarify',
      queryParameters: {'query': query},
    );

    if (response.statusCode == 200) {
      final data = response.data['data'] as List;
      return data.cast<String>();
    }

    return [];
  }
}
